Fix unhandled TransformType in qMRMLVirtualRealityTransformWidget

The following code should be updated to address the warnings listed below:

https://github.com/KitwareMedical/SlicerVirtualReality/blob/13bef667e04f2ea13435ff55ef85b521c45bf8d3/VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx#L202-L213

### Warnings

```
/path/to/SlicerVirtualReality/VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx:202:10: warning: enumeration value ‘TrackedDeviceClass_Invalid’ not handled in switch [-Wswitch]
  202 |   switch (d->TransformType)
      |          ^
/path/to/SlicerVirtualReality/VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx:202:10: warning: enumeration value ‘TrackedDeviceClass_TrackingReference’ not handled in switch [-Wswitch]
/path/to/SlicerVirtualReality/VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx:202:10: warning: enumeration value ‘TrackedDeviceClass_DisplayRedirect’ not handled in switch [-Wswitch]
/path/to/SlicerVirtualReality/VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx:202:10: warning: enumeration value ‘TrackedDeviceClass_Max’ not handled in switch [-Wswitch]
/path/to/SlicerVirtualReality/VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx: In member function ‘void qMRMLVirtualRealityTransformWidget::updateWidgetFromMRML()’:
/path/to/SlicerVirtualReality/VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx:250:10: warning: enumeration value ‘TrackedDeviceClass_Invalid’ not handled in switch [-Wswitch]
  250 |   switch (d->TransformType)
      |          ^
/path/to/SlicerVirtualReality/VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx:250:10: warning: enumeration value ‘TrackedDeviceClass_TrackingReference’ not handled in switch [-Wswitch]
/path/to/SlicerVirtualReality/VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx:250:10: warning: enumeration value ‘TrackedDeviceClass_DisplayRedirect’ not handled in switch [-Wswitch]
/path/to/SlicerVirtualReality/VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx:250:10: warning: enumeration value ‘TrackedDeviceClass_Max’ not handled in switch [-Wswitch]
```

--- a/VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx
+++ b/VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx
@@ -210,6 +210,10 @@ void qMRMLVirtualRealityTransformWidget::onButtonClicked()
     case vr::TrackedDeviceClass_Controller:
       d->VRViewNode->SetControllerTransformsUpdate(!d->VRViewNode->GetControllerTransformsUpdate());
       break;
+    default:
+      qCritical() << Q_FUNC_INFO << ": Unable to handle controller button click due to unknown transform type:"
+        << d->TransformType;
+      return;
   }
 
   updateWidgetFromMRML();
@@ -339,6 +343,10 @@ void qMRMLVirtualRealityTransformWidget::updateWidgetFromMRML()
 
         }
       }
+      break;
+    default:
+      qCritical() << Q_FUNC_INFO << ": Unable to update transform widget due to unknown transform type:"
+        << d->TransformType;
       break;
   }
 }

